A79-non-per-call-metrics-architecture.md

```diff
@@ -0,0 +1,405 @@
+
+##### Go
+
+(To be filled)
+
+### Metric Instrument Naming Conventions
+
+gRPC metrics follow the same general
+[guidelines](https://opentelemetry.io/docs/specs/semconv/general/metrics/#general-guidelines)
+as OpenTelemetry. The following additional guidelines apply -
+
+*   Metrics defined by gRPC should go under the `grpc.` namespace.
+*   Third-party plugins to gRPC should NOT use the `grpc.` namespace.
+*   Newer metrics should follow the same style as existing metrics. * Metrics
+    from load-balancing policies should be nested under `grpc.lb.`, e.g.
+    `grpc.lb.pick_first.`.
+*   Metrics from resolvers should be nested under `grpc.resolver.`, e.g.
+    `grpc.resolver.dns.`.
+*   Metrics from transports should be nested under `grpc.transport.`, e.g.,
+    `grpc.transport.http2.`
+
+### Metric Stability Guidelines and Guarantees
+
+From a stability perspective, metrics exported by gRPC will be in one of the
+following categories -
+
+*   On by default
+*   Off by default
+    *   Experimental
+    *   Non-Experimental
+
+Experimental metrics should always be off-by-default. gRPC reserves the right to
+change or remove these metrics. Documentation for the given metric determines
+whether the metric is experimental or non-experimental. An alternative approach
+of using an `experimental` namespace was considered, but not preferred since
+de-experimentalization of the metric would result in disruption of the
+corresponding metrics data (timeseries).
+
+New metrics should start as experimental. Once they are stable, they can
+optionally be made on-by-default. The distinction between on-by-default and
+off-by-default metrics allows gRPC to support metrics that may be desirable in
+some cases but too expensive to collect by default.
+
+### Changes to gRPC OpenTelemetry Plugin
+
+Optional Labels - Users should be able to configure gRPC OpenTelemetry plugins
+to record the additional optional labels. One suggested approach of implementing
+this is for the plugin instance to store a bit set for each instrument where
+each bit denotes whether the plugin should record the label in the corresponding
+position of the set of optional labels.
+
+API to enable/disable metrics - Users should be able to override the default
+behavior.
+
+Channel Scoped OpenTelemetry Plugins - API should be provided based on the
+approach chosen from the `Scoping Stats Plugins for Channels` section.
+
+#### C++
+
+The following API will be added to `OpenTelemetryPluginBuilder` introduced in
+[A66: OpenTelemetry Metrics](A66-otel-stats.md).
+
+```c++
+class OpenTelemetryPluginBuilder {
+  // Records \a optional_label_key on all metrics that provide it.
+  void AddOptionalLabel(absl::string_view optional_label_key);
+  // Set scope filter to choose which channels are recorded by this plugin.
+  // Server-side recording remains unaffected.
+  OpenTelemetryPluginBuilder& SetChannelScope(
+      absl::AnyInvocable<bool(ChannelScope /*scope*/) const> scope_filter);
```

Comment:
I think this will need to be `StatsPlugin::ChannelScope`.

Reply:
added a using statement